### ISSUE TITLE
Code Style Guidelines:  Internals

### ISSUE THREAD

**Issue by tomspilman** [dotnet/corefx#1216]
So the existing coding style guidelines...

https://github.com/dotnet/corefx/wiki/Coding-style

... do not give guidance with regard to internal members.

In the existing code base we see several different styles in use for field members:

https://github.com/dotnet/corefx/blob/master/src/System.Diagnostics.Process/src/System/Diagnostics/ThreadInfo.cs#L15
```
    internal sealed class ThreadInfo
    {
        internal int _threadId;
        internal int _processId;
```

https://github.com/dotnet/corefx/blob/master/src/System.Linq.Parallel/src/System/Linq/Parallel/Utils/Shared.cs#L18
```
    internal class Shared<T>
    {
        internal T Value;
```

https://github.com/dotnet/corefx/blob/master/src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlChildEnumerator.cs#L10
```
    internal sealed class XmlChildEnumerator : IEnumerator
    {
        internal XmlNode container;
        internal XmlNode child;
        internal bool isFirst;
```

It would be useful to the core project and other open source projects that are reusing your guidelines and the codeformatter tool (https://github.com/dotnet/codeformatter) to have this more well defined.

**Comment by weshaggard** [comment]
I just updated https://github.com/dotnet/corefx/wiki/Coding-style to call out that we use follow the same naming guideline for both internal and private fields. So the first one is correct and the second two are not correct. I know the xml code was put out before we ran the formatting tool on it and we have a issue to format it all correctly. I suspect the same is true of the System.Linq.Parellel code. We will get to doing a larger pass for cleaning up code styles and formatting at some point. 

Thanks for the getting this clarified.
